x-pack/plugins/alerting/server/rules_client/audit_events.ts

```diff
@@ -88,6 +90,7 @@ const eventTypes: Record<RuleAuditAction, EcsEventType> = {
   rule_resolve: 'access',
   rule_update: 'change',
   rule_bulk_edit: 'change',
+  rule_bulk_delete: 'deletion',
```

Comment:
New audit events should be added to the docs here: https://github.com/elastic/kibana/blob/main/docs/user/security/audit-logging.asciidoc

However, I'm a bit confused by the bulk event that you added. There shouldn't be any bulk events in the audit log but instead individual events, one for each resource:

> Operations that are performed on multiple resources (bulk operations) should be logged as separate events, one for each resource.

More details and context can be found in the developer guidelines for audit logging:
https://github.com/elastic/kibana/blob/main/dev_docs/key_concepts/audit_logging.mdx

Reply:
We are using this new property to audit around our authorization functionality like we did for the `bulk edit`. However, you are right we are missing the audit log to tell which rule has been deleted. We will add that and test to show that we did that correctly. Thank you!

Reply:
@thomheymann , here I added audit logs for every successful or not rule deletion: 487cfdcd1773bd903b03bb710ade44c8b52c8f7b

Reply:
@XavierM Is it possible for users to be authorised to bulk delete but not authorised to delete individual rules (or the other way around)?

Reply:
@thomheymann based on code I do not see how it's possible. So I would say no.

Reply:
> @thomheymann based on code I do not see how it's possible. So I would say no.

In that case I would suggest to change the `rule_bulk_delete` to `rule_delete` to make it easier for security analysts to search the audit log and for consistency with the rest of the audit events.

Reply:
I agree @thomheymann.
What do you think about this: e01d35cb36d1a4681b259f5496f5d8e57d210b0e?
Does docs looks good?

Reply:
Thank you, that looks much better. 

One remaining issue is that write operations should be logged before carrying out the operation so the outcome should be "unknown" instead of "success". Have a read through the "When should an event be logged?" section for context and explanation: https://github.com/elastic/kibana/blob/main/dev_docs/key_concepts/audit_logging.mdx#when-should-an-event-be-logged

An example of the correct flow can be seen in the `bulkCreate` method here:
https://github.com/elastic/kibana/blob/main/x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts#L191-L201

Reply:
Thanks for guidance, @thomheymann!
27fc33abaa07972afd0a04cec3b4f3481c5a05ab

Reply:
Brilliant, thanks for taking the time to fix all of these!